src/main/java/network/brightspots/rcv/Tabulator.java

```diff
@@ -851,13 +852,24 @@ private void recordSelectionForCastVoteRecord(
           selectedCandidate,
           cvr.getFractionalTransferValue());
       if (config.isTabulateByPrecinctEnabled()) {
-        precinctTallyTransfers
-            .get(cvr.getPrecinct())
-            .addTransfer(
-                currentRound,
-                cvr.getCurrentRecipientOfVote(),
-                selectedCandidate,
-                cvr.getFractionalTransferValue());
+        if (precinctIds.size() == 0) {
+          Logger.severe("\"Tabulate by Precinct\" enabled, but CVRs don't list precincts.");
+          throw new TabulationAbortedException(false);
+        }
+
+        String precinctId = cvr.getPrecinct();
```

Comment:
Should we also check if this is `null` first, or is there no need?

Reply:
There's no need, it'll fail gracefully: whether it's null or an unknown value, it'll get caught in the same way below.